### PR TITLE
Fix/article CRUD

### PR DESCRIPTION
**Pull request by amaurymn**

### CODE REVIEW

File: src/Controllers/AdminController/ArticleController.php [hunk 1]
```diff
@@ -5,31 +5,109 @@
...
+     * @throws \ReflectionException
+     * @throws \Twig\Error\LoaderError
+     * @throws \Twig\Error\RuntimeError
+     * @throws \Twig\Error\SyntaxError
+     */
     public function executeShowArticleAdd()
     {
-        $article = new Article(['admin_id' => 1, 'id' => 1]);
+        $article = new Article(['admin_id' => 1]);
+        $article->setCreatedAt(new \DateTime());
```
Comment (Tutt-J) [inline src/Controllers/AdminController/ArticleController.php:40]: You must add this directly on Timestamplable entity

File: src/Controllers/AdminController/ArticleController.php [hunk 2]
```diff
@@ -5,31 +5,109 @@
...
+    /**
+     * @throws \ReflectionException
+     * @throws \Twig\Error\LoaderError
+     * @throws \Twig\Error\RuntimeError
+     * @throws \Twig\Error\SyntaxError
+     */
     public function executeShowArticleAdd()
     {
-        $article = new Article(['admin_id' => 1, 'id' => 1]);
+        $article = new Article(['admin_id' => 1]);
```
Comment (Tutt-J) [inline src/Controllers/AdminController/ArticleController.php:39]: Call it when form is submit

File: src/Controllers/AdminController/ArticleController.php [hunk 3]
```diff
@@ -5,31 +5,109 @@
...
+        $article->setCreatedAt(new \DateTime());
+        $article->setUpdatedAt(new \DateTime());
 
-        if($this->isFormSubmit('publish')) {
+        if ($this->isFormSubmit('publish')) {
```
Comment (Tutt-J) [inline src/Controllers/AdminController/ArticleController.php:43]: refactore order with function like $form->isValid()

File: src/Controllers/AdminController/ArticleController.php [hunk 4]
```diff
@@ -5,31 +5,109 @@
...
 
+    /** @var array|bool */
+    private $formErrors;
+
+    /**
+     * @throws \Twig\Error\LoaderError
+     * @throws \Twig\Error\RuntimeError
+     * @throws \Twig\Error\SyntaxError
+     */
     public function executeShowArticleList()
```
Comment (Tutt-J) [inline src/Controllers/AdminController/ArticleController.php:22]: Rename executeReadList

File: src/Controllers/AdminController/ArticleController.php [hunk 5]
```diff
@@ -5,31 +5,109 @@
...
+        ]);
     }
 
+    /**
+     * @throws \ReflectionException
+     * @throws \Twig\Error\LoaderError
+     * @throws \Twig\Error\RuntimeError
+     * @throws \Twig\Error\SyntaxError
+     */
     public function executeShowArticleAdd()
```
Comment (Tutt-J) [inline src/Controllers/AdminController/ArticleController.php:37]: Rename it executeCreate

File: src/Controllers/AdminController/ArticleController.php [hunk 6]
```diff
@@ -5,31 +5,109 @@
...
+        ]);
     }
 
+    /**
+     * @throws \ReflectionException
+     * @throws \Twig\Error\LoaderError
+     * @throws \Twig\Error\RuntimeError
+     * @throws \Twig\Error\SyntaxError
+     */
     public function executeShowArticleEdit()
```
Comment (Tutt-J) [inline src/Controllers/AdminController/ArticleController.php:65]: Rename executeUpdate

File: src/Controllers/AdminController/ArticleController.php [hunk 7]
```diff
@@ -5,31 +5,109 @@
...
     {
-        $this->render('@admin/articleEdit.html.twig');
+        $getArticle  = (new ArticleManager())->findOneBy(['id' => $this->params['articleId']]);
```
Comment (Tutt-J) [inline src/Controllers/AdminController/ArticleController.php:67]: Rename $article

File: src/Controllers/AdminController/ArticleController.php [hunk 8]
```diff
@@ -5,31 +5,109 @@
...
+        $editArticle = new Article(['admin_id' => 1, 'id' => $getArticle['id']]);
```
Comment (Tutt-J) [inline src/Controllers/AdminController/ArticleController.php:68]: Remove this you must use $article

File: src/Controllers/AdminController/ArticleController.php [hunk 9]
```diff
@@ -5,31 +5,109 @@
...
+            }
+        }
+
+        $this->render('@admin/articleEdit.html.twig', [
+            'article' => $getArticle,
+            'errors'  => $this->formErrors
+        ]);
+    }
+
+    public function executeDeleteArticle(): void
```
Comment (Tutt-J) [inline src/Controllers/AdminController/ArticleController.php:89]: executeDelete

File: src/Controllers/AdminController/ArticleController.php [hunk 10]
```diff
@@ -5,31 +5,109 @@
...
+    {
+        $getArticle = (new ArticleManager())->findOneBy(['id' => $this->params['articleId']]);
```
Comment (Tutt-J) [inline src/Controllers/AdminController/ArticleController.php:91]: rename $article

File: src/Controllers/AdminController/ArticleController.php [hunk 11]
```diff
@@ -5,31 +5,109 @@
...
+        $article    = new Article(['id' => $getArticle['id']]);
```
Comment (Tutt-J) [inline src/Controllers/AdminController/ArticleController.php:92]: rename deleteArticle or remove

File: src/Core/Manager.php [hunk 12]
```diff
@@ -59,7 +58,7 @@ public function findBy(array $where = [], array $order = [], int $limit = null,
      */
     public function findOneBy(array $where = [], array $order = [])
     {
-        return $this->findBy($where, $order, 0, 1)[0];
+        return $this->findBy($where, $order, 0, 1);
```
Comment (Tutt-J) [inline src/Core/Manager.php:61]: You must leave [0] you must administrate case where nothing is found

**Review by Tutt-J — CHANGES_REQUESTED** [review]

File: src/Core/Manager.php (hunk 12 above)
Comment (amaurymn) [reply src/Core/Manager.php:61]: `Undefined offset` si je supprime [0]
Il n'y a pas d'index car la fonction `findBy` utilise `fetch` si il y a 1 résultat et `fetchAll` si il y en a plusieurs.

fonction findBy
`return ($stmt->rowCount() > 1) ? $stmt->fetchAll() : $stmt->fetch();`

**Pull request merged**